client/core/core.go

```diff
@@ -4646,6 +5023,10 @@ func (c *Core) handleReconnect(host string) {
 	}
 }
 
+func (dc *dexConnection) broadcastingConnect() bool {
+	return atomic.LoadUint32(&dc.reportingConnects) == 1
+}
```

Comment:
Don't know if this is just me, but it wasn't immediately clear to me what purpose the `dc.reportingConnects` field served. I had to scan through the usages in code to realise it is basically used to differentiate between temporary and non-temporary dcs. I want to recommend renaming to `temporary` and this method to `isTemporary()`.